FODO_Test_Lattice/Phase_Error_Propagation/phase_propagation.py

```diff
@@ -0,0 +1,289 @@
+from dataclasses import dataclass, fields
+from pathlib import Path
+from typing import Tuple, Union
+
+import numpy as np
+import pandas as pd
+import tfs
+from cpymad.madx import Madx
+from matplotlib import colors as mc
+from matplotlib import pyplot as plt
+from uncertainties import ufloat
+
+RNG = np.random.default_rng(27837)
+FloatOrArray = Union[float, np.ndarray, pd.Series]
+
+
+# Classes ----------------------------------------------------------------------
+
+@dataclass 
+class PropagableBoundaryConditions:
+    """Store boundary conditions with error for propagating."""
+    alpha: ufloat = None
+    beta: ufloat = None
+
+    @staticmethod
+    def as_tuple(value: ufloat) -> Tuple[float, float]:
+        return (value.nominal_value, value.std_dev)
+
+
+@dataclass 
+class MadXBoundaryConditions:
+    """Store all boundary conditions for a Mad-X twiss."""
+    alfx: float = None
+    alfy: float = None
+    betx: float = None
+    bety: float = None
+    dx: float = None
+    dy: float = None
+    dpx: float = None
+    dpy: float = None
+    wx: float = None
+    wy: float = None
+    dphix: float = None
+    dphiy: float = None
+    r11: float = None
+    r12: float = None
+    r21: float = None
+    r22: float = None
+
+    def as_dict(self):
+        return {f.name: getattr(self, f.name) for f in fields(self) if getattr(self, f.name) is not None}
+
+
+# Main -------------------------------------------------------------------------
+
+def create_realizations(output: Path, n: int = 100) -> pd.DataFrame:
+    """Main function to generate the phase advance realizations.
+    The mean and std values are hardcoded and choosen to have a final phase advance 
+    of around 0.25 (far away from 0.5) in units of 2 Pi.
+
+    Args:
+        output (Path): Path to save the output tfs-file.
+        n (int, optional): Number of realizations to generate. Defaults to 100.
+    """
+    madx = generate_fodo()
+    madx.twiss(chrom=True)
+
```

Comment:
add comment # define boundary conditions with uncertainty using uncertainties.ufloat(value, err) or similar

Reply:
I would say that is pretty clear from the class-name, also given the doc-string of the function.